lib/media/adaptation_set_criteria.js

```diff
@@ -280,6 +278,24 @@ shaka.media.PreferenceBasedCriteria = class {
     });
   }
 
+  /**
+   * Filter Variants by channelCount.
+   *
+   * @param {!Array.<shaka.extern.Variant>} variants
+   * @param {number} channelCount
+   * @return {!Array.<shaka.extern.Variant>}
+   * @private
+   */
+  static filterVariantsByAudioChannelCount_(variants, channelCount) {
+    return variants.filter((variant) => {
```

Comment:
This seems like a pretty big difference in behavior?
The old function used `preferredAudioChannelCount` as a maximum channel count; this takes it as a required exact value.
The old function usually filtered away ignored variants without audio and channelCounts; this one passes them through untouched.
The old function would keep the lowest channelsCount if everything had too many channels; this won't, so it will end up returning an empty array in that case.
And so on...

I do think, having examined the original method, that the JSDoc for `preferredAudioChannelCount` describes what the config value does very poorly; what you have here might be more accurate to what the JSDoc implies.

However, the AdaptationSetCriteria functionality is a part of the core player, and not the UI. So I don't think something tagged `feat(UI)` is the right place for a significant functionality change for it.

I think I'd ask you to move this into a separate PR. And maybe make part of that PR be setting the default value of `preferredAudioChannelCount` to 0 so that no filtering happens by default, since this new mechanism is a bit stricter than the old one.

Reply:
I moved it to https://github.com/shaka-project/shaka-player/pull/5875